pyvista/plotting/colors.py

```diff
@@ -876,13 +883,16 @@ def __hash__(self):  # pragma: no cover
         return hash((self._red, self._green, self._blue, self._opacity))
 
     def __getitem__(self, item):
-        """Support indexing for backward compatibility."""
+        """Support indexing the float RGBA representation for backward compatibility."""
+        for i, cnames in enumerate(self.CHANNEL_NAMES):
+            if item in cnames:
+                item = i
```

Comment:
I would start with an `isinstance(item, (str, int, np.integer))` check, then only do this loop if `item` is a string.